### PR TITLE
Documented 2 new APIs for .NET Core 2.2

### PR DESCRIPTION
**Pull request by rpetrusha**

### CODE REVIEW

**Comment by rpetrusha** [comment]
It seems to me that it is worth adding this information, @maira @vancem and @brianrob I've revised the topic.

I noticed that Remarks sections belonging to an enumeration member are not rendered (whereas they were previously on MSDN). Because of that, I've moved all remarks that apply to individual enumeration members to the enumeration type's Remarks section. Is this a known issue, @mairaw and @dend? I've checked a few other enum types, and the content dropout appears in each case.

**Comment by mairaw** [comment]
Yes, I believe I gave this feedback in the past @rpetrusha. I can't recall what I heard back.

File: xml/System.Diagnostics.Tracing/EventWrittenEventArgs.xml [hunk 1]
```diff
@@ -66,20 +92,9 @@
         <ReturnType>System.Guid</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>Gets the activity ID on the thread that the event was written to.</summary>
+        <summary>Gets the activity ID on the thread that the event was written to. See the Remarks section for more information. </summary>
```
Comment (mairaw) [inline xml/System.Diagnostics.Tracing/EventWrittenEventArgs.xml]: I think we don't usually add this given IntelliSense won't have remarks.

Comment (rpetrusha) [reply xml/System.Diagnostics.Tracing/EventWrittenEventArgs.xml]: Good point. I've removed them, @mairaw.

File: xml/System.Diagnostics.Tracing/EventWrittenEventArgs.xml [hunk 2]
```diff
@@ -29,7 +29,33 @@
...
+
+Starting with .NET Core 2.2, <xref:System.Diagnostics.Tracing.EventListener> objects can subscribe to native runtime events (such as GC, JIT, and threadpool events) in addition to events emitted by <xref:System.Diagnostics.Tracing.EventSource> objects. In previous versions of .NET Core and all versions of .NET Framework, the thead ID and timestamp can be gathered from the environment, because they are dispatched synchronously on the same thread that emitted them. Not all native runtime events can be dispatched synchronously, however. Some events, such as GC events, are emitted when managed thread execution is suspended. These events are buffered in native code and are dispatched by a dispatcher thread once managed code can execute again. Because these events are buffered, the environment cannot be used to reliably retrieve the thread ID and timestamp. Because of this, starting with .NET Core 2.2, thread ID and timestamp information are available as members of the `EventWrittenEventArgs` class.
+
+### RelatedActivityId property
+
+The `RelatedActivityId` prooerty is a child of the `ActivityId` property. An activity ID can be associated with multiple related activity IDs.
```
Comment (vancem) [inline xml/System.Diagnostics.Tracing/EventWrittenEventArgs.xml]: prooerty  -> property

Comment (vancem) [inline xml/System.Diagnostics.Tracing/EventWrittenEventArgs.xml]:
```suggestion
The `RelatedActivityId` property is a child of the `ActivityId` property. An activity ID can be associated with multiple related activity IDs.
```

File: xml/System.Diagnostics.Tracing/EventWrittenEventArgs.xml [hunk 3]
```diff
@@ -29,7 +29,33 @@
...
+    <remarks>
+          <format type="text/markdown"><![CDATA[  
+  
+## Remarks  
+
+Whenever an event is dispatched to an <xref:System.Diagnostics.Tracing.EventListener>, the <xref:System.Diagnostics.Tracing.EventListener.OnEventWritten&2A?displayProperty=nameWithType> callback method is invoked. It is passed an `EventWrittenEventArgs` instance that contains information associated with the event. 
+
+The following sections contain additional information about individual `EventWrittenEventArgs` properties.
+
+### ActivityId property
```
Comment (vancem) [inline xml/System.Diagnostics.Tracing/EventWrittenEventArgs.xml]: You can remove this section on the ActivityId property.   It is subsumed by the suggestion above (which is that all fields are only valid during the callback.    I would not worry about ActivityId changing during the callback, it is very unlikely to be relevant, but I leave that to you.

File: xml/System.Diagnostics.Tracing/EventWrittenEventArgs.xml [hunk 4]
```diff
@@ -593,8 +601,8 @@
         <ReturnType>System.DateTime</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
-        <value>To be added.</value>
+        <summary>Gets the date and time of the event. (.NET Core 2.2 and later only.)</summary>
+        <value>The date and time of the event.</value>
```
Comment (vancem) [inline xml/System.Diagnostics.Tracing/EventWrittenEventArgs.xml]:
```suggestion
        <value>The time (and date) when the event was logged.   The system tries to be as precise as possible.  Typically
        the time is known to a granularity well under a millisecond.   </value>
```

File: xml/System.Diagnostics.Tracing/EventWrittenEventArgs.xml [hunk 5]
```diff
@@ -66,20 +92,9 @@
...
-## Remarks  
- Retrieving the value of the <xref:System.Diagnostics.Tracing.EventWrittenEventArgs.ActivityId%2A> property accesses thread local storage. This has the following implications:  
-  
--   The <xref:System.Diagnostics.Tracing.EventWrittenEventArgs.ActivityId%2A> field is valid only for the duration of the <xref:System.Diagnostics.Tracing.EventListener.OnEventWritten%2A?displayProperty=nameWithType> callback.  
-  
--   If the activity ID is changed during the callback, the activity ID returned by the <xref:System.Diagnostics.Tracing.EventWrittenEventArgs.ActivityId%2A> property is invalid.  
-  
- ]]></format>
-        </remarks>
+        <remarks>To be added.</remarks>
```
Comment (vancem) [inline xml/System.Diagnostics.Tracing/EventWrittenEventArgs.xml]:
```suggestion
        <remarks>Using the System.Activity class, threads can be marked as having an activity associated 
        with them and this returns the Activity ID of the thread that logged the event.    Note that threads
        do not have to have an activity, in which case this property returns Guid.Empty</remarks>
```

File: xml/System.Diagnostics.Tracing/EventWrittenEventArgs.xml [hunk 6]
```diff
@@ -29,7 +29,33 @@
...
+
+### OSThreadId and TimeStamp properties 
+
+Starting with .NET Core 2.2, <xref:System.Diagnostics.Tracing.EventListener> objects can subscribe to native runtime events (such as GC, JIT, and threadpool events) in addition to events emitted by <xref:System.Diagnostics.Tracing.EventSource> objects. In previous versions of .NET Core and all versions of .NET Framework, the thead ID and timestamp can be gathered from the environment, because they are dispatched synchronously on the same thread that emitted them. Not all native runtime events can be dispatched synchronously, however. Some events, such as GC events, are emitted when managed thread execution is suspended. These events are buffered in native code and are dispatched by a dispatcher thread once managed code can execute again. Because these events are buffered, the environment cannot be used to reliably retrieve the thread ID and timestamp. Because of this, starting with .NET Core 2.2, thread ID and timestamp information are available as members of the `EventWrittenEventArgs` class.
+
+### RelatedActivityId property
```
Comment (vancem) [inline xml/System.Diagnostics.Tracing/EventWrittenEventArgs.xml]: I would remove this section on RelatedActivityId.   I have put the important part into the remarks for the property.  I don't think it is importnat to indicate that one activity may have more than one RelatedActivityId (although it is true).    For what it is worth, a particular event can only have at most one RelatedActivityId.

**Comment by vancem** [comment]
I have left a number of suggestions.   If you take them, then I am happy.

File: xml/System.Diagnostics.Tracing/EventWrittenEventArgs.xml (hunk 6 above)
Comment (rpetrusha) [reply xml/System.Diagnostics.Tracing/EventWrittenEventArgs.xml]: Remarks for an individual property aren't rendering, @vancem, which is why I've added detail on individual properties to the general remarks section (a build design bug). I'll incorporate the remarks you've added in the general remarks section.

**Comment by rpetrusha** [comment]
I've incorporated your feedback, @vancem, if you'd like to take another look.

**Pull request merged**